### ISSUE TITLE
[YAML] "on" (and related) should not be scoped as constant in key position

### ISSUE THREAD

**Issue by djspiewak** [sublimehq/Packages#2329]
```yaml
  on: on
# ^^ entity.name.tag.yaml - constant
#     ^^ constant.language.boolean.yaml

```

This test fails at present because the `on` *key* is scoped the same as the `on` *value*. This also holds true for all the other boolean constants (e.g. `true`), and really *all* constants (e.g. `42`).

This comes up increasingly frequently these days because `on` is a mandatory key in any GitHub Actions workflow definition.

**Comment by michaelblyons** [comment]
See #1602 for short, very related conversation.

**Comment by FichteFoll** [comment]
It kind of annoys me as well, but it's a duplicate of #1602. I'll add a comment regarding my current opinion there.

**Comment by wbond** [comment]
This has been resolved - see #1602
